### PR TITLE
Feat: Allow users to select another user as their out-of-office replacement

### PR DESCRIPTION
**Pull request by hamza221**
* part of: [#40398](https://github.com/nextcloud/server/issues/40398)

## Summary

| b | a |
|--------|--------|
| image | image | 

## TODO

- [x] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/Controller/OutOfOfficeController.php [hunk 1]
```diff
@@ -93,6 +93,7 @@ public function getOutOfOffice(string $userId): DataResponse {
 			'lastDay' => $data->getLastDay(),
 			'status' => $data->getStatus(),
 			'message' => $data->getMessage(),
+			'replacementUserId' => $data->getReplacementUserId(),
```
Comment (nickvergessen) [inline apps/dav/lib/Controller/OutOfOfficeController.php:96]: if we also add the replacement user displayname, talk UI and clients don't need to make another request just to get that.

File: apps/dav/lib/Migration/Version1031Date20240610134258.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
```
Comment (kesselb) [inline apps/dav/lib/Migration/Version1031Date20240610134258.php:6]:
```suggestion
 * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
```

File: apps/dav/lib/Migration/Version1031Date20240610134258.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version1031Date20240610134258 extends SimpleMigrationStep {
+	/**
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
```
Comment (kesselb) [inline apps/dav/lib/Migration/Version1031Date20240610134258.php:24]:
```suggestion
```

File: apps/dav/src/components/AbsenceForm.vue [hunk 4]
```diff
@@ -39,13 +54,16 @@
...
+import NcSelect from '@nextcloud/vue/dist/Components/NcSelect.js'
 import NcDateTimePickerNative from '@nextcloud/vue/dist/Components/NcDateTimePickerNative.js'
 import { generateOcsUrl } from '@nextcloud/router'
 import { getCurrentUser } from '@nextcloud/auth'
+import debounce from 'debounce'
 import axios from '@nextcloud/axios'
 import { formatDateAsYMD } from '../utils/date.js'
 import { loadState } from '@nextcloud/initial-state'
 import { showError, showSuccess } from '@nextcloud/dialogs'
+import {  Type as ShareTypes } from '@nextcloud/sharing'
```
Comment (kesselb) [inline apps/dav/src/components/AbsenceForm.vue:66]:
```suggestion
import { Type as ShareTypes } from '@nextcloud/sharing'
```

**Review by kesselb — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]

**Pull request merged**